app/main/helpers/framework_helpers.py

```diff
@@ -7,3 +7,7 @@ def get_latest_live_framework(all_frameworks, framework_type):
         return latest
     except ValueError:  # max of empty iterable
         return None
+
+
+def get_lots_by_slug(framework_data):
+    return dict((lot['slug'], lot) for lot in framework_data['lots'])
```

Comment:
Slightly more optimal and clear to use a dict comprehension here, I think:
`{lot['slug']: lot for lot in framework_data['lots']}`